notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "<i>Actually, these plots don't really help us, because, looking on them, we can only catch some common \"trends\", which describe global changes in some products sales.<br>For example, we can be sure that on the 1th January every year number of sales is equal to 0 (actually, it is necessary to check, but I think it is obvious). <br> Also we can say that the number of sales in general has a positive dynamic (for example, 'automotive', 'bread/bakery', 'grocery i', 'personal care', ...).<br>Some of the product types have a negative dynamic (such as lingerie).<br>Some goods have very interesting sales distribution (books, produce, froxen foods, ladieswear, ...), and we can't say right now, what is the reason of that.<br>Also some goods families have seasonal increase in sales('school and office supplies', 'liquor, wine, beer', 'grocery ii', 'frozen foods'). </i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 110))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "    plt.subplot(17, 2, i+1)\n",
+    "    average_sales = train_data[train_data['family'] == family_value].groupby('date').mean()['sales']\n",
+    "    trend = average_sales.rolling(\n",
+    "        window=365,\n",
+    "        center=True,\n",
+    "        min_periods=183,\n",
+    "    ).mean()\n",
+    "    ax = average_sales.plot(alpha=0.5)\n",
+    "    ax = trend.plot(ax=ax, linewidth=3)\n",
+    "    plt.title(family_value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 70))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data['family'] == value].groupby([\"date\"]).mean()\n",
+    "  sns.distplot(current_value_data['sales'], color='g', bins=100, hist_kws={'alpha': 0.4});\n",
+    "  plt.title(value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>These plots show that, roughly speaking, sales distributions of all families divided on two parts: <br>1) Normal or close to normal (such as 'automotive', 'bread/bakery', 'cleaning', 'eggs', 'grocery', 'lingerie', ...). Interesting fact that most of distributions from this category have right asymmetry (asymmetry coefficient is positive). The prove is below.<br>2) Distribution, where the biggest density is concentrated in zero or near zero. Other data is distributed differently (some values such as 'home and kitchen i', 'ladieswear' have something like normal distributions). It means that such goods categories aren't essential for people, that is why a number of sales during the day mostly is equal to 0.</i>"
```

Comment:
> 1) Normal or close to normal (such as 'automotive', 'bread/bakery', 'cleaning', 'eggs', 'grocery', 'lingerie', ...)

Can you say something common about all these categories? Can we name them as first consumption goods?

> It means that such goods categories aren't essential for people, that is why a number of sales during the day mostly is equal to 0.

For these categories I would suggest exclude zeros and re-build distributions, probably something interesting will become visible (but maybe not - who knows :))

Reply:
"Can we name them as first consumption goods?"

Most of such categories are first consumption goods. I can name these categories as exceptions: 'automotive', 'seafood'.